[BUG] "error adding symbols: DSO missing from command line" when install rTorrent

### What happened?

Steps to reproduce:

1. Install swizzin in a fresh system by using: 
`bash <(wget -O- -q https://raw.githubusercontent.com/liaralabs/swizzin/master/setup.sh) --user aaa --pass xxx`

2. got error :
> ERROR   Something went wrong while making rtorrent
        Please consult the above and/or check the log (less -R +G /root/logs/swizzin.log)
INFO    Installing rutorrent
DOCS    Further reference: https://swizzin.ltd/applications/rutorrent
ERROR   ruTorrent is a GUI for rTorrent, which doesn't appear to be installed. Exiting.
        Please consult the above and/or check the log (less -R +G /root/logs/swizzin.log)

Here is the ERROR log screenshot.
[https://images4.imagebam.com/e4/90/44/MEMS49Q_o.png](https://images4.imagebam.com/e4/90/44/MEMS49Q_o.png)

Expected results:
solve the error.


### Swizzin commit

361c5e7

### What OS are you using?

Ubuntu 20.04 (focal)

### What architecture is your OS?

amd64

### Relevant logs and output

```plaintext
mv -f .deps/thread_base.Tpo .deps/thread_base.Po
g++ -DHAVE_CONFIG_H -I. -I..  -I. -I..   -w -O1    -MT thread_worker.o -MD -MP -MF .deps/thread_worker.Tpo -c -o thread_
worker.o thread_worker.cc
mv -f .deps/thread_worker.Tpo .deps/thread_worker.Po
rm -f libsub_root.a
ar cru libsub_root.a command_download.o command_dynamic.o command_events.o command_file.o command_ip.o command_helpers.o
 command_groups.o command_local.o command_logging.o command_network.o command_peer.o command_throttle.o command_tracker.
o command_scheduler.o command_ui.o control.o globals.o option_parser.o signal_handler.o thread_base.o thread_worker.o
ar: `u' modifier ignored since `D' is the default (see `U')
ranlib libsub_root.a
/bin/bash ../libtool  --tag=CXX   --mode=link g++  -w -O1      -o rtorrent main.o libsub_root.a ui/libsub_ui.a core/libs
ub_core.a display/libsub_display.a input/libsub_input.a rpc/libsub_rpc.a utils/libsub_utils.a  -Wl,-Bsymbolic-functions
-lncursesw -ltinfo  -lcppunit  -L/usr/local/lib -lcurl -lgcc -lgcc_s -lc -lgcc -lgcc_s -llber -lldap /usr/local/lib/libc
ares.a /usr/lib/x86_64-linux-gnu/libssl.so /usr/lib/x86_64-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libnghttp2.s
o /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libzstd.so /usr/lib/x86_64-linux-gnu/libpsl.so /usr/lib/x8
6_64-linux-gnu/libssh2.so -ltorrent  -L/usr/lib   -lxmlrpc_server -lxmlrpc  -lxmlrpc_xmlparse -lxmlrpc_xmltok -lxmlrpc_u
til -lpthread
libtool: link: g++ -w -O1 -o rtorrent main.o -Wl,-Bsymbolic-functions /usr/lib/x86_64-linux-gnu/libssl.so /usr/lib/x86_6
4-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libnghttp2.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux
-gnu/libzstd.so /usr/lib/x86_64-linux-gnu/libpsl.so /usr/lib/x86_64-linux-gnu/libssh2.so  libsub_root.a ui/libsub_ui.a c
ore/libsub_core.a display/libsub_display.a input/libsub_input.a rpc/libsub_rpc.a utils/libsub_utils.a -lncursesw -ltinfo
 -lcppunit -L/usr/local/lib -lcurl -lc -lgcc -lgcc_s -llber -lldap /usr/local/lib/libcares.a /usr/lib/libtorrent.so -L/u
sr/lib -lxmlrpc_server -lxmlrpc -lxmlrpc_xmlparse -lxmlrpc_xmltok -lxmlrpc_util -lpthread
/usr/bin/ld: /tmp/rtorrent.VAvChG.ltrans7.ltrans.o: undefined reference to symbol 'inflateEnd'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libz.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:563: rtorrent] Error 1
make[3]: Leaving directory '/tmp/rtorrent/src'
make[2]: *** [Makefile:634: all-recursive] Error 1
make[2]: Leaving directory '/tmp/rtorrent/src'
make[1]: *** [Makefile:452: all-recursive] Error 1
make[1]: Leaving directory '/tmp/rtorrent'
make: *** [Makefile:384: all] Error 2
[Thu Jul 13 15:50:20] ERROR     Something went wrong while making rtorrent
[Thu Jul 13 15:50:20]   Please consult the above and/or check the log (less -R +G /root/logs/swizzin.log)
[Thu Jul 13 15:50:20] INFO      Installing rutorrent
[Thu Jul 13 15:50:20] DOCS      Further reference: https://swizzin.ltd/applications/rutorrent
[Thu Jul 13 15:50:20] ERROR     ruTorrent is a GUI for rTorrent, which doesn't appear to be installed. Exiting.
[Thu Jul 13 15:50:20]   Please consult the above and/or check the log (less -R +G /root/logs/swizzin.log)
[Thu Jul 13 15:50:20] INFO      Installing deluge
[Thu Jul 13 15:50:20] DOCS      Further reference: https://swizzin.ltd/applications/deluge
[Thu Jul 13 15:50:20] ...       Choosing Deluge versions
[Thu Jul 13 15:50:20] ...       Building libtorrent-rasterbar
[Thu Jul 13 15:50:20] ...       Installing boost
--2023-07-13 15:50:20--  https://boostorg.jfrog.io/artifactory/main/release/1.75.0/source/boost_1_75_0.tar.gz
Resolving boostorg.jfrog.io (boostorg.jfrog.io)... 34.210.93.108, 54.186.43.133, 35.164.107.174, ...
Connecting to boostorg.jfrog.io (boostorg.jfrog.io)|34.210.93.108|:443... connected.
HTTP request sent, awaiting response... 302
```

Done. the same error. :(

> /usr/bin/ld: /tmp/rtorrent.W26nD7.ltrans7.ltrans.o: undefined reference to symbol 'inflateEnd'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libz.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:568: rtorrent] Error 1
make[3]: Leaving directory '/tmp/rtorrent/src'
make[2]: *** [Makefile:639: all-recursive] Error 1
make[2]: Leaving directory '/tmp/rtorrent/src'
make[1]: *** [Makefile:457: all-recursive] Error 1
make[1]: Leaving directory '/tmp/rtorrent'
make: *** [Makefile:389: all] Error 2
[Fri Jul 14 02:37:05] ERROR     Something went wrong while making rtorrent

--- a/sources/functions/curl
+++ b/sources/functions/curl
@@ -25,7 +25,7 @@ build_cares() {
     cd /tmp/cares
     rm CMakeCache.txt >> $log 2>&1
 
-    cmake . -D CARES_STATIC=ON -D CARES_SHARED=OFF -D CARES_STATIC_PIC=ON -D CMAKE_BUILD_TYPE:STRING="Release" -D CMAKE_C_FLAGS_RELEASE:STRING="-w -O3 -flto=\"$(nproc)\" -march=native" >> ${log} 2>&1 || {
+    cmake . -D CARES_STATIC=ON -D CARES_SHARED=ON -D CARES_STATIC_PIC=ON -D CMAKE_BUILD_TYPE:STRING="Release" -D CMAKE_C_FLAGS_RELEASE:STRING="-w -O3 -flto=\"$(nproc)\" -march=native" >> ${log} 2>&1 || {
         echo_error "There was an error configuring c-ares! Please check the log for more info"
         cd /tmp
         rm -rf /tmp/cares >> $log 2>&1
@@ -58,7 +58,7 @@ build_curl() {
     cd /tmp/curl-debian-bullseye-backports
     rm CMakeCache.txt >> $log 2>&1
 
-    cmake . -D ENABLE_ARES=ON -D ENABLE_WEBSOCKETS=ON -D CURL_LTO=ON -D CURL_USE_OPENSSL=ON -D CURL_USE_OPENLDAP=ON -D CURL_BROTLI=ON -D CURL_ZSTD=ON -D CURL_USE_LIBPSL=ON -D USE_NGHTTP2=ON -D BUILD_SHARED_LIBS=OFF -D CMAKE_BUILD_TYPE:STRING="Release" -D CMAKE_C_FLAGS_RELEASE:STRING="-w -O3 -flto=\"$(nproc)\" -march=native" >> ${log} 2>&1 || {
+    cmake . -D ENABLE_ARES=ON -D ENABLE_WEBSOCKETS=ON -D CURL_LTO=ON -D CURL_USE_OPENSSL=ON -D CURL_USE_OPENLDAP=ON -D CURL_BROTLI=ON -D CURL_ZSTD=ON -D CURL_USE_LIBPSL=ON -D USE_NGHTTP2=ON -D BUILD_SHARED_LIBS=ON -D CMAKE_BUILD_TYPE:STRING="Release" -D CMAKE_C_FLAGS_RELEASE:STRING="-w -O3 -flto=\"$(nproc)\" -march=native" >> ${log} 2>&1 || {
         echo_error "There was an error configuring curl! Please check the log for more info"
         cd /tmp
         rm -rf /tmp/curl-* >> $log 2>&1

